test/win-bats/collection.xml

```diff
@@ -2159,7 +2159,7 @@
     call :verify_line * r "  my-error-code[: ] Error signalled .*"
 	</case>
 
-	<case name="Error in SUT should not be caught by default (XQuery)">
+	<case ifdef="SAXON12_INITIAL_ISSUES_FIXED" name="Error in SUT should not be caught by default (XQuery)">
```

Comment:
Saxon 12 does not report the error code `my-error-code` but does include the error message. Here is a small query that demonstrates a similar behavior difference between Saxon 11 and 12.
```
xquery version "3.1";

declare namespace xs = "http://www.w3.org/2001/XMLSchema";

declare function local:fcn($result as item()*) as item()* {
$result
};

(: the query body :)
let $result as item()* := error(xs:QName('my-error-code'))
return local:fcn($result)
```
Outputs:
```
java -cp "C:\Program Files\Saxonica\SaxonHE11.5J\saxon-he-11.5.jar" net.sf.saxon.Query  -q:repro-issue-error-code.xq
Error on line 10 column 43 of repro-issue-error-code.xq:
  my-error-code  Error signalled by application call on error()
<?xml version="1.0" encoding="UTF-8"?>Query failed with dynamic error: Error signalled by application call on error()


java -cp "C:\Program Files\Saxonica\SaxonHE12.0J\saxon-he-12.0.jar" net.sf.saxon.Query  -q:repro-issue-error-code.xq
<?xml version="1.0" encoding="UTF-8"?>net.sf.saxon.trans.UncheckedXPathException: Error signalled by application call on error()
        at net.sf.saxon.value.SingletonClosure.iterate(SingletonClosure.java:81)
        at net.sf.saxon.expr.LocalVariableReference$LocalVariableReferenceElaborator.lambda$elaborateForPush$1(LocalVariableReference.java:214)
        at net.sf.saxon.expr.instruct.UserFunction.process(UserFunction.java:712)
        at net.sf.saxon.expr.UserFunctionCall$UserFunctionCallElaborator.lambda$elaborateForPush$5(UserFunctionCall.java:846)
        at net.sf.saxon.query.XQueryExpression.processQuery(XQueryExpression.java:499)
        at net.sf.saxon.query.XQueryExpression.run(XQueryExpression.java:473)
        at net.sf.saxon.s9api.XQueryEvaluator.run(XQueryEvaluator.java:464)
        at net.sf.saxon.Query.runQuery(Query.java:975)
        at net.sf.saxon.Query.doQuery(Query.java:444)
        at net.sf.saxon.Query.main(Query.java:106)
Caused by: net.sf.saxon.functions.Error$UserDefinedXPathException: Error signalled by application call on error()
        at net.sf.saxon.functions.Error.error(Error.java:67)
        at net.sf.saxon.functions.Error.call(Error.java:121)
        at net.sf.saxon.expr.SystemFunctionCall$SystemFunctionCallElaborator.lambda$elaborateForPull$1(SystemFunctionCall.java:608)
        at net.sf.saxon.value.SingletonClosure.asItem(SingletonClosure.java:112)
        at net.sf.saxon.value.SingletonClosure.iterate(SingletonClosure.java:72)
        ... 9 more
Fatal error during query: net.sf.saxon.trans.UncheckedXPathException: Error signalled by application call on error()
```
I see the latter behavior with Saxon versions 12.0 through 12.2.

I don't see a problem if I augment the small query by using a try/catch block or if I simplify the small query by not calling `local:fcn`.

Reply:
Logged as https://saxonica.plan.io/issues/6081 along with the omission of error code XPDY0002 in another test case.

Reply:
This is Case 1 in https://saxonica.plan.io/issues/6116 . (The earlier issue 6081 was fixed in Saxon 12.3 for the small, self-contained queries but not for the actual query seen here.)